pronotepy/clients.py

```diff
@@ -148,7 +148,9 @@ def _login(self) -> bool:
             log.info('login failed')
             return False
 
-    def get_week(self, date: datetime.date):
+    def get_week(self, date: Union[datetime.date, datetime.datetime]):
+        if isinstance(date, datetime.datetime):
+            return 1 + int((date.date() - self.start_day).days / 7)
```

Comment:
Maybe leave it as it was. This is quite nit picky but an `isinstance` check is quite expensive. This method is only used internally and mostly with `datetime.date`, the caller can easily convert it from `datetime.datetime` outside.

Reply:
the "isinstance" function does not cost that much. I join you the profiling of different calls

Reply:
```
import cProfile

cProfile.run('client.lessons(datetime.datetime.now())')
```
[restats.txt](https://github.com/bain3/pronotepy/files/7117694/restats.txt)

```
   ncalls  tottime  percall  cumtime  percall filename:lineno(function)
     1136    0.000    0.000    0.000    0.000 {built-in method builtins.isinstance}
```

```
import cProfile

cProfile.run('client.lessons(datetime.datetime.now(), datetime.datetime.now() + datetime.timedelta(weeks=52))')
```
[stat_52_weeks.txt](https://github.com/bain3/pronotepy/files/7117700/stat_52_weeks.txt)

```
   ncalls  tottime  percall  cumtime  percall filename:lineno(function)
    46200    0.005    0.000    0.007    0.000 {built-in method builtins.isinstance}
```